### ISSUE TITLE
New TiffFile release breaking our builds

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#3991]
It appears that the [latest tifffile release](https://pypi.org/project/tifffile/#history) (2019.7.2) has broken our builds by deprecating the `pages=` keyword argument. Example broken build:

https://travis-ci.org/scikit-image/scikit-image/jobs/555062286#L2646

```pytb
            if 'pages' in kwargs:
                raise TypeError(
>                   "the TiffFile 'pages' argument is no longer supported.\n\n"
                    "Use TiffFile.asarray(keys=[...]) to read image data "
                    "from specific pages.\n")
E               TypeError: the TiffFile 'pages' argument is no longer supported.
E               
E               Use TiffFile.asarray(keys=[...]) to read image data from specific pages.
../../venv/lib/python3.5/site-packages/tifffile/tifffile.py:2209: TypeError
```

Examining the release notes, it seems those were discontinued for a while, but either there were no visible warnings or we missed them altogether.

Possible solutions:

- remove testing of `pages=` from our tests
- wrap the new API so that users of `skimage.io.imread(..., pages=...)` can happily carry on with their lives.

I see pluses and minuses to both.

**Comment by jni** [comment]
I think this is pretty critical to fix quickly (it breaks most builds in our matrix) so if @scikit-image/core express their preferences, I'll make a PR.

**Comment by hmaarrfk** [comment]
Follow upstream and remove pages from our tests.

**Comment by jni** [comment]
@cgohlke the error message says to use `asarray(keys=[...]`), but when I try that (should work with `skimage.io.imread(filepath, keys=[...])`, as seen by the code snippet below), I get an unexpected keyword argument 'keys' TypeError:

https://github.com/scikit-image/scikit-image/blob/6dfc9ee0a38bd35076a46444b043f87a0f252522/skimage/io/_plugins/tifffile_plugin.py#L39-L49

What's the correct way forward here?

**Comment by jni** [comment]
Update: looking at the definition of `asarray`, it turns out to be `key=`. Probably worth pushing a new tifffile release to update the error message, but in the meantime I've got the fix working, PR forthcoming.

**Comment by cgohlke** [comment]
> skimage.io.imread(filepath, keys=[...])

should read `skimage.io.imread(filepath, key=[...])`

**Comment by jni** [comment]
@cgohlke thanks! But as pointed above, the raised TypeError incorrectly advises the use of the `keys=` keyword argument.

**Comment by cgohlke** [comment]
I understand. It is a misprint in the error message.

**Comment by cgohlke** [comment]
> It appears that the latest tifffile release (2019.7.2) has broken our builds by deprecating the pages= keyword argument. 

The `pages` parameter was removed from the `TiffFile` constructor in version `2017.9.29`.

**Comment by emmanuelle** [comment]
I agree with @hmaarrfk , change our tests in order to use `key` instead.

**Comment by emmanuelle** [comment]
Maybe also we should have more examples with io in the doc...

**Comment by sciunto** [comment]
I would suggest to  do this once pooch is inplace.

**Comment by sciunto** [comment]
Closed by #3992
